k2/csrc/ragged_ops.cu

```diff
@@ -1038,6 +1026,7 @@ Array1<int32_t> GetTransposeReordering(Ragged<int32_t> &src, int32_t num_cols) {
   const int32_t *value_data = src.values.Data();
   int32_t n = src.values.Dim();
   Array1<int32_t> ans = Range(context, n, 0);
+  if (n == 0) return ans;
```

Comment:
mgpu will throw cuda error if ans is emtpy, @csukuangfj it would be great if you can add random tests (especially empty case) for all our mgpu usages. (IIRC, we use it for `arc_sort` and `TransposeReorder`?)

Reply:
Will add it.